pastas/recharge.py

```diff
@@ -98,13 +99,13 @@ def __init__(self):
         RechargeBase.__init__(self)
         self.nparam = 1
 
-    def get_init_parameters(self, name="recharge"):
+    def get_init_parameters(self, name: Optional[str] = "recharge") -> Type[DataFrame]:
         parameters = DataFrame(
             columns=["initial", "pmin", "pmax", "vary", "name"])
         parameters.loc[name + "_f"] = (-1.0, -2.0, 0.0, True, name)
         return parameters
 
-    def simulate(self, prec, evap, p, **kwargs):
+    def simulate(self, prec: pstAL, evap: pstAL, p: pstAL, **kwargs) -> pstAL:

```

Comment:
I now copy the NumPy type `ArrayLike` to define our own array_like (pstAL) type. Since we use a lot of NumPy arrays in Pastas and they were documented either as `np.array`, `array`, `np.ndarray`, `array_like` etc. I thought it would be best to define one type for all (and for now copy it from NumPy's definition).